lib/lotus/action/cookie_jar.rb

```diff
@@ -131,6 +137,7 @@ def extract(env)
         hash   = env[COOKIE_HASH_KEY] ||= {}
         string = env[HTTP_HEADER]
 
+        return if hash.has_key? RACK_SESSION_KEY
```

Comment:
@AlfonsoUceda Isn't this `nil` return value dangerous for the caller? It expects a Hash, but get `nil` if already the session was already set.